### PR TITLE
Update sigma-rust to 0.14.0-alpha-a3940fc

### PR DESCRIPTION
**Pull request by arobsn**
Updates sigma-rust to latest version via the following changes:

 - Updated `ergo-lib-wasm-nodejs` and `ergo-lib-wasm-browser` to `0.14.0-alpha-a3940fc`
 - Added parsing to all necessary `to_json()` result via `JSONBigInt.parse()`

### CODE REVIEW

File: packages/yoroi-extension/app/api/ergo/lib/transactions/ErgoExternalTxSignRequest.js [hunk 1]
```diff
@@ -348,7 +348,5 @@ export function ergoTxEqual(
   req1: RustModule.SigmaRust.UnsignedTransaction,
   req2: RustModule.SigmaRust.UnsignedTransaction,
 ): boolean {
-  const tx1 = req1.to_json();
-  const tx2 = req2.to_json();
-  return JSON.stringify(tx1) === JSON.stringify(tx2);
+  return req1.to_json() === req2.to_json();
```
Comment (zuzunker) [inline packages/yoroi-extension/app/api/ergo/lib/transactions/ErgoExternalTxSignRequest.js:351]: Sweet

File: packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js [hunk 2]
```diff
@@ -454,7 +455,7 @@ export function signTransaction(request: {|
       RustModule.SigmaRust.ErgoBoxes.from_boxes_json([]), // TODO: not supported by sigma-rust
     );
 
-  const json = signedTx.to_json();
+  const json = JSONBigInt.parse(signedTx.to_json());
```
Comment (zuzunker) [inline packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js:458]: @anon-br , if I remember correctly previously used version of the library were returning output values and asset values as strings. At least this is the contract for the dapp-connector API. So now that you are parsing JSON with the BigInt library - you might have to manually convert those specific fields to strings as they now will be parsed as numbers or BigNumbers

File: packages/yoroi-extension/app/api/jormungandr/lib/state-fetch/mockNetwork.js [hunk 3]
```diff
@@ -575,7 +576,7 @@ export function toRemoteByronTx(
   const signedTx = RustModule.WalletV2.SignedTransaction
     .from_bytes(Buffer.from(request.signedTx, 'base64'));
   const hash = signedTx.id();
-  const transaction = signedTx.to_json().tx;
+  const transaction = JSONBigInt.parse(signedTx.to_json()).tx;
```
Comment (zuzunker) [inline packages/yoroi-extension/app/api/jormungandr/lib/state-fetch/mockNetwork.js]: Should not change parsing in this case. Here the `signedTx` is produced by `RustModule.WalletV2`, which is related to Cardano. Only values produced by `RustModule.SigmaRust` need updating.

**Review by zuzunker — CHANGES_REQUESTED** [review]

File: packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js (hunk 2 above)
Comment (zuzunker) [reply packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js:458]: @anon-br , yeah, if you check the return type of this function: `SignedRequest`

The fields `.outputs.*value` and `.outputs.*assets.*amount` must be returned as strings:

image

But now that you are using `JSONBigInt` - they will be parsed as `number | BigNumber` so in the return object you must explicitly convert it to `String(v)`

File: packages/yoroi-extension/chrome/extension/ergo-connector/api.js [hunk 4]
```diff
@@ -374,7 +375,7 @@ export async function connectorSignTx(
       dataBoxesToSpend,
     );
   debug('signedTx', '', signedTx);
-  return signedTx.to_json();
+  return JSONBigInt.parse(signedTx.to_json());
```
Comment (zuzunker) [inline packages/yoroi-extension/chrome/extension/ergo-connector/api.js]: Same thing here, check the function return type: `ErgoTxJson`

Output values and output asset amounts must be strings:

image

So gotta do explicit conversion

**Review by zuzunker — CHANGES_REQUESTED** [review]

File: packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js (hunk 2 above)
Comment (arobsn) [reply packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js:458]: Hmn gotcha

File: packages/yoroi-extension/app/api/jormungandr/lib/state-fetch/mockNetwork.js (hunk 3 above)
Comment (arobsn) [reply packages/yoroi-extension/app/api/jormungandr/lib/state-fetch/mockNetwork.js]: Oh, right. Didn't realize that.

**Review by zuzunker — CHANGES_REQUESTED** [review]
@anon-br, why there's one new import like added in this file but no any other changes?

image

**Comment by arobsn** [comment]
Oh, I forget to remove this after change requested [here](https://github.com/Emurgo/yoroi-frontend/pull/2391#discussion_r709521747), sorry :)

Already removed in the new commit.

**Review by zuzunker — APPROVED** [review]

**Review by zuzunker — COMMENTED** [review]
/check

**Review by zuzunker — APPROVED** [review]

**Pull request merged**